[Doc] Add schema configuration to conf file and reference documentation (#13983)
apache/pulsar: site2/docs/reference-configuration.md

|athenzDomainNames| Supported Athenz provider domain names(comma separated) for authentication  ||
|exposePreciseBacklogInPrometheus| Enable expose the precise backlog stats, set false to use published counter and consumed counter to calculate, this would be more efficient but may be inaccurate. |false|
|schemaRegistryStorageClassName|The schema storage implementation used by this broker.|org.apache.pulsar.broker.service.schema.BookkeeperSchemaStorageFactory|

> |isSchemaValidationEnforced| Whether to enable schema validation, when enabled, if a producer without a schema attempts to produce to a topic with schema, the producer will be rejected and disconnected.|false|

Reviewer: same

|exposePreciseBacklogInPrometheus| Enable expose the precise backlog stats, set false to use published counter and consumed counter to calculate, this would be more efficient but may be inaccurate. |false|
|schemaRegistryStorageClassName|The schema storage implementation used by this broker.|org.apache.pulsar.broker.service.schema.BookkeeperSchemaStorageFactory|
|isSchemaValidationEnforced|Enforce schema validation on following cases: if a producer without a schema attempts to produce to a topic with schema, the producer will be failed to connect. PLEASE be carefully on using this, since non-java clients don't support schema. If this setting is enabled, then non-java clients fail to produce.|false|
|isAllowAutoUpdateSchemaEnabled|Allow schema to be auto updated at broker level|true|

> |schemaCompatibilityStrategy|Schema compatibility check strategy, supported value: `ALWAYS_COMPATIBLE`,`ALWAYS_INCOMPATIBLE`,`BACKWARD`,`BACKWARD_TRANSITIVE`,`FORWARD`,`FORWARD_TRANSITIVE`,`FULL`,`FULL_TRANSITIVE`|FULL|

Reviewer 2: My suggestion:
Replace ", supported value:" with ". Available values:".

> |systemTopicSchemaCompatibilityStrategy|The schema compatibility strategy to use for system topics, supported value: `ALWAYS_COMPATIBLE`,`ALWAYS_INCOMPATIBLE`,`BACKWARD`,`BACKWARD_TRANSITIVE`,`FORWARD`,`FORWARD_TRANSITIVE`,`FULL`,`FULL_TRANSITIVE`|ALWAYS_COMPATIBLE|

Reviewer 2: My suggestion:
Replace ", supported value:" with ". Available values:".

|athenzDomainNames| Supported Athenz provider domain names(comma separated) for authentication  ||
|exposePreciseBacklogInPrometheus| Enable expose the precise backlog stats, set false to use published counter and consumed counter to calculate, this would be more efficient but may be inaccurate. |false|
|schemaRegistryStorageClassName|The schema storage implementation used by this broker.|org.apache.pulsar.broker.service.schema.BookkeeperSchemaStorageFactory|
|isSchemaValidationEnforced| Whether to enable schema validation, when schema validation is enabled, if a producer without a schema attempts to produce the message to a topic with schema, the producer is rejected and disconnected.|false|
|isAllowAutoUpdateSchemaEnabled|Allow schema to be auto updated at broker level.|true|

> |schemaCompatibilityStrategy| The schema compatibility strategy at broker level, see [here](schema-evolution-compatibility.md#schema-compatibility-check-strategy) for available values.|FULL|

Author: Use URL to avoid duplicate descriptions of available values.

|haProxyProtocolEnabled | Enable or disable the [HAProxy](http://www.haproxy.org/) protocol. |false|
|bookieId | If you want to custom a bookie ID or use a dynamic network address for a bookie, you can set the `bookieId`. <br><br>Bookie advertises itself using the `bookieId` rather than the `BookieSocketAddress` (`hostname:port` or `IP:port`).<br><br> The `bookieId` is a non-empty string that can contain ASCII digits and letters ([a-zA-Z9-0]), colons, dashes, and dots. <br><br>For more information about `bookieId`, see [here](http://bookkeeper.apache.org/bps/BP-41-bookieid/).|/|
| maxTopicsPerNamespace | The maximum number of persistent topics that can be created in the namespace. When the number of topics reaches this threshold, the broker rejects the request of creating a new topic, including the auto-created topics by the producer or consumer, until the number of connected consumers decreases. The default value 0 disables the check. | 0 |
|schemaRegistryStorageClassName|The schema storage implementation used by this broker.|org.apache.pulsar.broker.service.schema.BookkeeperSchemaStorageFactory|

> |isSchemaValidationEnforced| Whether to enable schema validation, when enabled, if a producer without a schema attempts to produce to a topic with schema, the producer will be rejected and disconnected.|false|

Reviewer: same

> |isAllowAutoUpdateSchemaEnabled|Allow schema to be auto updated at broker level|true|

Reviewer: ```suggestion
|isAllowAutoUpdateSchemaEnabled|Allow schema to be auto updated at broker level.|true|
```

Reviewer: @nodece this comment is overlooked, any thoughts?

P.S. please resolve conflicts in this PR before requesting a review, thanks.

> |isSchemaValidationEnforced|Enforce schema validation on following cases: if a producer without a schema attempts to produce to a topic with schema, the producer will be failed to connect. PLEASE be carefully on using this, since non-java clients don't support schema. If this setting is enabled, then non-java clients fail to produce.|false|

Reviewer 2: ```suggestion
|isSchemaValidationEnforced|Enforce schema validation. As a result, when a producer without a schema attempts to produce a topic with schema, the producer will be disconnected.<br />Note: If enabled, non-java clients cannot produce data because they don't support schema.|false|
```

Author: Thanks, I update this description, could you review it again?

> |schemaCompatibilityStrategy|Schema compatibility check strategy. Available values: `ALWAYS_COMPATIBLE`,`ALWAYS_INCOMPATIBLE`,`BACKWARD`,`BACKWARD_TRANSITIVE`,`FORWARD`,`FORWARD_TRANSITIVE`,`FULL`,`FULL_TRANSITIVE`|FULL|

Reviewer: ```suggestion
|schemaCompatibilityStrategy|Schema compatibility check strategy. Available values: `ALWAYS_COMPATIBLE`,`ALWAYS_INCOMPATIBLE`,`BACKWARD`,`BACKWARD_TRANSITIVE`,`FORWARD`,`FORWARD_TRANSITIVE`,`FULL`,`FULL_TRANSITIVE`|`FULL`|
```

> |systemTopicSchemaCompatibilityStrategy|The schema compatibility strategy to use for system topics. Available values: `ALWAYS_COMPATIBLE`,`ALWAYS_INCOMPATIBLE`,`BACKWARD`,`BACKWARD_TRANSITIVE`,`FORWARD`,`FORWARD_TRANSITIVE`,`FULL`,`FULL_TRANSITIVE`|ALWAYS_COMPATIBLE|

Reviewer: ```suggestion
|systemTopicSchemaCompatibilityStrategy|The schema compatibility strategy to use for system topics. Available values: `ALWAYS_COMPATIBLE`,`ALWAYS_INCOMPATIBLE`,`BACKWARD`,`BACKWARD_TRANSITIVE`,`FORWARD`,`FORWARD_TRANSITIVE`,`FULL`,`FULL_TRANSITIVE`|`ALWAYS_COMPATIBLE`|
```
